python/tskit/trees.py

```diff
@@ -3831,15 +3863,19 @@ def __reversed__(self):
         return self
 
     def __next__(self):
-        if self.forward:
-            self.more_trees = self.more_trees and self.tree.next()
-        else:
-            self.more_trees = self.more_trees and self.tree.prev()
-        if not self.more_trees:
-            raise StopIteration()
+        while True:
```

Comment:
This looks correct, but I think we'd need a good bit of testing to check what happens when we have runs of empty trees in the middle. Although, come to think of it I don't think we can have two adjacent empty trees, so maybe this while look could be replace with something simpler?